src/components/Common/SearchByMultipleFields.tsx

```diff
@@ -0,0 +1,259 @@
+  buttonClassName?: string;
+}
+
+type EventType = {
+  value: string;
+  target?: { value: string };
+};
+
+const SearchByMultipleFields: React.FC<SearchByMultipleFieldsProps> = ({
+  options,
+  onSearch,
+  initialOption,
+  className,
+  inputClassName,
+  buttonClassName,
+}) => {
+  const { t } = useTranslation();
+  const [selectedOption, setSelectedOption] = useState<SearchOption>(
+    initialOption || options[0],
+  );
+  const [searchValue, setSearchValue] = useState(selectedOption.value || "");
+  const [open, setOpen] = useState(false);
+  const inputRef = useRef<HTMLInputElement>(null);
+  const [focusedIndex, setFocusedIndex] = useState(0);
+  const [error, setError] = useState<string | undefined | boolean>();
+  useEffect(() => {
+    const handleKeyDown = (e: KeyboardEvent) => {
+      if (e.key === "/" && document.activeElement !== inputRef.current) {
+        e.preventDefault();
+        setOpen(true);
+      } else if (e.key === "ArrowDown") {
+        setFocusedIndex((prevIndex) =>
+          prevIndex === options.length - 1 ? 0 : prevIndex + 1,
+        );
+      } else if (e.key === "ArrowUp") {
+        setFocusedIndex((prevIndex) =>
+          prevIndex === 0 ? options.length - 1 : prevIndex - 1,
+        );
+      } else if (e.key === "Enter") {
+        handleOptionChange(options[focusedIndex]);
+      }
+
+      if (e.key === "Escape") {
+        inputRef.current?.focus();
+        setOpen(false);
+      }
+
+      options.forEach((option) => {
+        if (e.key === "/") {
+          setOpen(true);
+        } else if (
+          e.key.toLocaleLowerCase() ===
+            option.shortcutKey.toLocaleLowerCase() &&
+          open
+        ) {
+          e.preventDefault();
+          handleOptionChange(option);
+        }
+      });
+    };
+
+    document.addEventListener("keydown", handleKeyDown);
+    return () => document.removeEventListener("keydown", handleKeyDown);
+  }, [focusedIndex, open]);
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  }, [selectedOption]);
+  const handleOptionChange = useCallback(
+    (option: SearchOption) => {
+      setSelectedOption(option);
+      setSearchValue(option.value || "");
+      setFocusedIndex(options.findIndex((op) => op.key === option.key));
+      setOpen(false);
+      inputRef.current?.focus();
+      setError(false);
+      onSearch(option.key, option.value);
+    },
+    [onSearch],
+  );
+
+  const handleSearchChange = useCallback(
+    (value: string) => {
+      setSearchValue(value);
+      onSearch(selectedOption.key, value);
+    },
+    [selectedOption, onSearch],
+  );
+
+  const handleKeyDown = useCallback(
+    (e: React.KeyboardEvent<HTMLInputElement>) => {
+      if (e.key === "Escape") {
+        setSearchValue("");
+        onSearch(selectedOption.key, "");
+      }
+      if (e.key === "/") {
+        e.preventDefault();
+        setOpen(true);
+      }
+    },
+    [selectedOption, onSearch],
```

Comment:
`handleKeyDown` is recreated on every render due to missing dependencies. 

Please fix this issue for all hooks in this file.